.changeset/small-knives-sparkle.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': minor
+---
+
+Throw better error when a dynamic endpoint without additional extensions is prerendered with `undefined` params. For example, `src/pages/blog/[slug].js` and `src/pages/blog/[...slug].js` can't have a `getStaticPaths` function that returns `undefined` for `slug`, Astro would generate both a `/blog` file and `/blog` directory (for thruthy slugs), causing a build error. To fix this, you can add an extension like `[...slug].json.js` to generate `/blog.json` instead.
```

Comment:
```suggestion
Throw better error when a dynamic endpoint without additional extensions is prerendered with `undefined` params.
```
I think for the changeset, we'd normally just stick to something like this? The change is "a better error." I don't *think* you need to explain in such detail?

If I'm wrong, ping me again and I'll fix a typo/run on sentence!  :smile: